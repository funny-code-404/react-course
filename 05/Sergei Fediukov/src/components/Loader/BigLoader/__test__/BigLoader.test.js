import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import { BigLoader } from '..'
configure({ adapter: new Adapter() });

describe('Should render correctly "ResultNavigation"', () => {
    const mockFn = jest.fn();

    const props = {
        civilizations: { isFetching: true },
        units: { isFetching: false },
        structures: { isFetching: false },
        technologies: { isFetching: false },
    }
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<BigLoader {...props} />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
    describe('loader should appear correctly', () => {
        it('component`s state should change ', () => {
            wrapper.setProps({
                civilizations: { isFetching: false }
            })
            expect(wrapper.find('.loading')).toHaveLength(0)
            expect(wrapper.find('span')).toHaveLength(0)
            expect(wrapper.find('img')).toHaveLength(0)
        })
    })
})