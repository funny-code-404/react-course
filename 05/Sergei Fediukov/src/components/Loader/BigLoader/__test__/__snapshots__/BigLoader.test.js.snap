// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly "ResultNavigation" to Match Snapshot 1`] = `
<div
  className="loading"
>
  <div>
    <span>
      Loading...
    </span>
  </div>
  <img
    alt="loading..."
    src="loading.gif"
  />
</div>
`;
